### PR TITLE
Execute quicktest for each SM driver

### PR DESCRIPTION
**Pull request by Wescoeur**
Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>

### CODE REVIEW

**Review by gduperrey — APPROVED** [review]

File: tests/storage/storage.py
Comment (stormi) [inline tests/storage/storage.py:74]: Maybe this could be a method of SR

**Comment by stormi** [comment]
The latest update candidate for xapi-tests will fix the behaviour of the `-default-sr` parameter of `quicktest`. You can install it on the hosts to test your PR.

File: tests/storage/storage.py
Comment (gthvn1) [reply tests/storage/storage.py:74]: I confirm that I run it with
```
sr.pool.master.ssh(['/opt/xensource/debug/quicktest', '-sr', sr.uuid])
```
So no need to have `get_default_sr` and `set_default_sr` anymore (at least when `-sr` will be available)

File: tests/storage/fsp/test_fsp_sr.py
Comment (benjamreis) [inline tests/storage/fsp/test_fsp_sr.py:6]: No longer needed.

**Review by benjamreis — CHANGES_REQUESTED** [review]

**Comment by stormi** [comment]
This PR has had a force push last month, but it's still named "[WIP]", and no re-review was requested. What's the current status?

**Comment by benjamreis** [comment]
IIRC we were waiting for new sources from XAPI to include the quicktest `--sr` option and from smapi to have the proper test filter applied.

**Comment by stormi** [comment]
Maybe I'm wrong, since you are the ones leading this development, but in my view I was waiting for you to tell me: this PR is good, you can review it.

**Comment by gthvn1** [comment]
Ooops sorry yes I back ported the patch to allow user to select the SR using the option. 
As this patch is the work of @Wescoeur I will let him modify the title and remove the WIP and check if everything is OK but for me it is ready for reviewing.

File: tests/storage/fsp/test_fsp_sr.py
Comment (benjamreis) [reply tests/storage/fsp/test_fsp_sr.py:6]: Still relevant

**Review by benjamreis — CHANGES_REQUESTED** [review]
also do we remove the `tests/quicktest` since now it is launched for each driver?

**Comment by Wescoeur** [comment]
>  As this patch is the work of @Wescoeur I will let him modify the title and remove the WIP and check if everything is OK but for me it is ready for reviewing.

Because I don't work on that, you can freely modify whatever you want. :)

**Review by benjamreis — APPROVED** [review]

File: tests/storage/storage.py
Comment (benjamreis) [reply tests/storage/storage.py:74]: IMHO this should stay as test method as it's only used in the `storage` scope.

Comment (stormi) [reply tests/storage/storage.py:74]: A method this simple and whose main input is a SR UUID, I would have put it on the SR class anyway (whose scope is storage anyway), and avoided a lot of imports in test files.

**Review by stormi — APPROVED** [review]
Next step, before merging: run the tests several times in a row to try to ensure they are stable.

**Comment by stormi** [comment]
Rebased on master branch

**Review by gthvn1 — APPROVED** [review]

**Pull request merged**
